### PR TITLE
style: add linting checks

### LINKED ISSUE #61: Consider adding new linting rules (e.g., add new line ad EOF).

### PR DESCRIPTION
**Pull request by LeonardHd**
added markdown, yaml, general file formatting checks to improve readability, version control handling, and consistency.

Closes #61

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,53 +102,53 @@ The samples seen below are the _only_ code that you need to write! The framework
...
-       assert "Job-123" == body["JobName"]
-       assert "version1" == body["Version"]
-
-       with pytest.raises(StopIteration):
-           next(activities)
-       ```
-   
-> See the [Examples](/examples) folder for more samples 
 
+1. Evaluate Pipelines and test the flow of activities given a specific input
```
Comment (arjendev) [inline README.md:106]: Something went wrong with the numbering here

File: docs/environment_setup/unit_test_setup.md [hunk 2]
```diff
@@ -1,82 +1,79 @@
...
 For more references: [Get Started Tutorial for Python in Visual Studio Code](https://code.visualstudio.com/docs/python/python-tutorial)
 
 1. Open your new Python project.
 
-
 ![image](/docs/environment_setup/images/new_project.png)
 
-
-2. Pip install the following libraries from the terminal - data-factory-testing-framework:
+1. Pip install the following libraries from the terminal - data-factory-testing-framework:
```
Comment (arjendev) [inline docs/environment_setup/unit_test_setup.md:21]: The numbering is also off here

File: docs/environment_setup/unit_test_setup.md [hunk 3]
```diff
@@ -1,82 +1,79 @@
...
 ![image](/docs/environment_setup/images/installing_poetry_power_shell.png)
 
-
-2. Add the bin to the PATH environment variable to the path
-
+1. Add the bin to the PATH environment variable to the path
 
 ![image](/docs/environment_setup/images/env_variable.png)
 
+1. Add the unit test framework for testing.
```
Comment (arjendev) [inline docs/environment_setup/unit_test_setup.md:73]: Same

**Review by arjendev — CHANGES_REQUESTED** [review]
Thanks for adding linting, super helpful! There's just a few numbering issues, other than that,  looks good!

**Review by jayachithra — APPROVED** [review]
Nice addition Leonard! Approved!

**Pull request merged**